File: springapp/src/main/java/com/examly/springapp/service/InvestmentInquiryService.java
package com.examly.springapp.service;
 
import java.util.List;
 
import com.examly.springapp.model.InvestmentInquiry;
 
public interface InvestmentInquiryService {
    InvestmentInquiry createInquiry(InvestmentInquiry investmentInquiry);
    InvestmentInquiry updateInquiry(Long inquiryId, InvestmentInquiry updatedInvestmentInquiry);
    List<InvestmentInquiry> getInquiriesByUserId(Long userId);
    List<InvestmentInquiry>getInquiriesByInvestmentId(Long investmentId);
    List<InvestmentInquiry>getUnresolvedInquiries();
    List<InvestmentInquiry>getInquiriesByPriority(String priority);
    List<InvestmentInquiry>getAllInquiries();
    InvestmentInquiry getInquiryById(Long inquiryId);
    Boolean deleteInquiry(Long inquiryId);
 
}

File: angularapp/src/app/services/investment-inquiry.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { InvestmentInquiry } from '../models/investment-inquiry.model';
import { Observable } from 'rxjs';
import { APP_URL } from 'src/global';

@Injectable({
  providedIn: 'root'
})
export class InvestmentInquiryService {

  constructor(private httpClient:HttpClient) { }

  addInquiry(inquiry:InvestmentInquiry):Observable<InvestmentInquiry>{
    return this.httpClient.post<InvestmentInquiry>(`${APP_URL}/inquiries`,inquiry);
  }

  getAllInquiries():Observable<InvestmentInquiry[]>{
    return this.httpClient.get<InvestmentInquiry[]>(`${APP_URL}/inquiries`);
  }

  getInquiriesByUserId(userId:number):Observable<InvestmentInquiry[]>{
    return this.httpClient.get<InvestmentInquiry[]>(`${APP_URL}/inquiries/user/${userId}`);
  }

  updateInquiry(inquiryId:number, inquiry:InvestmentInquiry):Observable<InvestmentInquiry>{
    return this.httpClient.put<InvestmentInquiry>(`${APP_URL}/inquiries/${inquiryId}`,inquiry);
  }

  deleteInquiry(inquiryId:number):Observable<void>{
    return this.httpClient.delete<void>(`${APP_URL}/inquiries/${inquiryId}`);
  }
}
